api/client/proto/authservice.proto

```diff
@@ -1359,6 +1363,42 @@ message CreateRegisterChallengeRequest {
     DeviceType DeviceType = 2 [ (gogoproto.jsontag) = "device_type" ];
 }
 
+// PaginatedResource represents one of the supported resources.
+message PaginatedResource {
+    // Resource is the resource itself.
+    oneof resource {
+        // DatabaseServer represents a DatabaseServer resource.
+        types.DatabaseServerV3 DatabaseServer = 1;
+        // ApplicationServer represents a AppServer resource.
+        types.AppServerV3 AppServer = 2;
+    }
+}
+
+// ListResourcesRequest defines a request to retrieve resources paginated. Only
+// one type of resource can be retrieved per request.
+message ListResourcesRequest {
+    // ResourceType is the resource that is going to be retrieved.
+    string ResourceType = 1;
+    // Namespace is the namespace of resources.
+    string Namespace = 2;
+    // Limit is the maximum amount of resources to retrieve.
+    int32 Limit = 3;
+    // StartKey is used to start listing resources from a specific spot. It
+    // should be set to the previous NextKey value if using pagination, or
+    // left empty.
+    string StartKey = 4;
```

Comment:
Can we set `gogoproto.jsontag` on these and other new fields like we do for all others here?